components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/SAMLAssertionDecryptor.java

```diff
@@ -0,0 +1,170 @@
+        return decrypter.decrypt(encryptedAssertion);
+    }
+
+    public static class DecryptorX509KeyStoreCredential implements X509Credential {
+
+        private PublicKey publicKey = null;
+        private PrivateKey privateKey = null;
+        private X509Certificate entityCertificate = null;
+
+        public DecryptorX509KeyStoreCredential(KeyStore keyStore, String publicCertAlias, String privateKeyAlias,
+                                               char[] privateKeyPassword) throws KeyStoreException,
+                UnrecoverableKeyException, NoSuchAlgorithmException {
+
+            readX509Credentials(keyStore, publicCertAlias, privateKeyAlias, privateKeyPassword);
+        }
+
+        public DecryptorX509KeyStoreCredential(InputStream keyStoreInputStream, char[] keyStorePassword,
+                                               String publicCertAlias, String privateKeyAlias,
+                                               char[] privateKeyPassword) throws UnrecoverableKeyException,
+                CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+
+            readX509Credentials(keyStoreInputStream, keyStorePassword, publicCertAlias,
+                    privateKeyAlias, privateKeyPassword);
+        }
+
+        protected void readX509Credentials(KeyStore keyStore, String publicCertAlias, String privateKeyAlias,
+                                           char[] privateKeyPassword) throws KeyStoreException,
+                UnrecoverableKeyException, NoSuchAlgorithmException {
+
+            entityCertificate = (X509Certificate) keyStore.getCertificate(publicCertAlias);
+            publicKey = entityCertificate.getPublicKey();
+            privateKey = (PrivateKey) keyStore.getKey(privateKeyAlias, privateKeyPassword);
+        }
+
+        protected void readX509Credentials(InputStream keyStoreInputStream, char[] keyStorePassword,
+                                           String publicCertAlias, String privateKeyAlias, char[] privateKeyPassword)
+                throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException,
+                UnrecoverableKeyException {
+
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(keyStoreInputStream, keyStorePassword);
+            readX509Credentials(keyStore, publicCertAlias, privateKeyAlias, privateKeyPassword);
+            if (keyStoreInputStream != null) {
+                keyStoreInputStream.close();
+            }
+        }
+
+        @Override
+        public PublicKey getPublicKey() {
+            return publicKey;
+        }
+
+        @Override
+        public PrivateKey getPrivateKey() {
+            return privateKey;
+        }
+
+        @Override
+        public X509Certificate getEntityCertificate() {
+            return entityCertificate;
+        }
+
+        // ********** Not implemented ************************************************************** //
```

Comment:
I don't think this comment is nice. Use standard method

Reply:
This comment is to indicate that the overridden methods under the comment are not actually overridden ones. THose are just there for the sake of implementing of X509Credential interface.
AFAIK this is the conventional way of adding that kind of comments.